weave

```diff
@@ -74,9 +74,15 @@ EOF
 }
 
 exec_remote() {
+    HOSTS_PATH=$(docker $DOCKER_CLIENT_ARGS run --rm --net=none \
+        -v /var/run/docker.sock:/var/run/docker.sock \
+        --entrypoint /bin/sh \
+        $EXEC_IMAGE -c 'docker inspect -f {{.HostsPath}} $(hostname)')
```

Comment:
This change introduces an extra call to Docker on every `weave xxx` invocation; it is necessary so we can bind-mount the containers directory and run `weavehosts` on `weave attach`.  The choice is to optimise run-speed of the proxy over `weave xxx`.

We run `docker` via `docker` in order to guarantee that there is one container we can inspect.

Perhaps a middle-ground approach would be to check if the command is 'attach' before obtaining `CONTAINERS_PATH` ?